app/src/lib/editors/launch.ts

```diff
@@ -23,9 +24,102 @@ export async function launchExternalEditor(
       { openPreferences: true }
     )
   }
-  if (editor.usesShell) {
-    spawn(`"${editorPath}"`, [`"${fullPath}"`], { shell: true })
+  if (
+    editor.editor === 'Visual Studio Code' ||
+    editor.editor === 'Visual Studio Code (Insiders)'
+  ) {
+    launchVisualStudioCode(editor, fullPath)
   } else {
-    spawn(editorPath, [fullPath])
+    const usesShell = editor.usesShell ? editor.usesShell : false
+    spwanExternalEditor(editorPath, fullPath, usesShell)
+  }
+}
+
+/**
+ * Open a given file or folder in Visual Studio Code.
+ *
+ * @param editor The external editor to launch.
+ * @param repoRootFolderPath The repository root folder to pass as an argument when launching the editor.
+ */
+function launchVisualStudioCode(
+  editor: FoundEditor,
+  repoRootFolderPath: string
+) {
+  const usesShell = editor.usesShell ? editor.usesShell : false
+  const useWorkspace = editor.useWorkspace ? editor.useWorkspace : false
+
+  if (!useWorkspace) {
+    spwanExternalEditor(editor.path, repoRootFolderPath, usesShell)
+  } else {
+    const workspacePattern = join(repoRootFolderPath, '*.code-workspace')
+    const glob = require('glob')
```

Comment:
Please lift this up to the start of the file and use the relevant `import` syntax:

```ts
import * as glob from 'glob'
```

This will allow you to simplify the signature for `glob` usage:

```ts
glob(workspacePattern, (error, files) => {
```